### PR TITLE
Scala3 Follow-Up #1: Fix Compiler Warnings

### PR DESCRIPTION
**Pull request by fm3**
- Fixes compiler warnings (mostly moving from implicit to using, and removing some unreachable match cases)
- re-enables failOnWarning in CI
- No reformatting yet, let’s do that separately for easier reviews.
- Note that some implicit parameters remain, since the compiler doesn’t warn about them. Those will probably still be replaced in later iterations.

### Steps to test:
- CI should be enough

### Issues:
- contributes to #9606 

------
- [x] Removed dev-only changes like prints and application.conf edits
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [x] Needs datastore update after deployment

### CODE REVIEW

File: app/controllers/Controller.scala [hunk 1]
```diff
@@ -2,8 +2,10 @@ package controllers
 
 import com.scalableminds.util.accesscontext.{AuthorizedAccessContext, DBAccessContext}
 import com.scalableminds.util.mvc.ExtendedController
+import com.scalableminds.util.tools.Fox
 import com.typesafe.scalalogging.LazyLogging
 import models.user.User
+import play.api.libs.json.{JsReadable, JsSuccess}
```
Comment (MichaelBuessemeyer) [inline app/controllers/Controller.scala]: why are these imports added. They seem unused

File: webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/skeleton/TreeValidator.scala [hunk 2]
```diff
@@ -24,7 +24,7 @@ object TreeValidator {
       _ <- checkAllTreeGroupIdsUsedExist(trees, treeGroups)
       _ <- checkAllNodesUsedInBranchPointsExist(trees, branchPoints)
       _ <- checkAllNodesUsedInCommentsExist(trees, comments)
-    } yield Full(())
+    } yield ()
```
Comment (MichaelBuessemeyer) [inline webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/skeleton/TreeValidator.scala:27]: this is an implicit conversion here now, correct? I thought we want to get rid off such magic and keep it more explicit. Can we maybe restore the `Full(())`?

**Review by MichaelBuessemeyer — APPROVED** [review]
Neat, thanks for the clean up.

Got 2 mini comments :D

File: webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/skeleton/TreeValidator.scala (hunk 2 above)
Comment (fm3) [reply webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/skeleton/TreeValidator.scala:27]: No, this was (wrongly) Box[Box[Unit]]. Now it’s just Box[Unit] :)

Comment (MichaelBuessemeyer) [reply webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/skeleton/TreeValidator.scala:27]: Ah I see :+1:
